mempool/src/pool.rs

```diff
@@ -385,10 +377,18 @@ where
             .filter_map(|input| input.outpoint().tx_id().get_tx_id().cloned())
             .filter_map(|id| self.store.txs_by_id.contains_key(&id).then_some(id))
             .collect::<BTreeSet<_>>();
+        let ancestor_ids =
+            TxMempoolEntry::unconfirmed_ancestors_from_parents(parents.clone(), &self.store)?;
+        let ancestors = ancestor_ids
+            .0
```

Comment:
It isn't related to your pull request, but I find this "index field access" somewhat confusing: I immediately want to check why it is a tuple and why only one field is utilized. Our `newtype` macro generates the `Deref` implementation, but `*ancestor_ids.into_iter()` breakes the type inference. Maybe we want to add a method such as `inner` to the `newtype` macro or is it only me who is frustrated in such cases?

Reply:
Opened an issue for this.

Reply:
It is just my personal opinion, so if everyone is OK with that, then I can live with that too. 🙃